core/services/ocr3/plugins/ccip/internal/reader/home_chain.go

```diff
@@ -0,0 +1,258 @@
+package reader
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+)
+
+type HomeChainConfigPoller struct {
+	homeChainReader types.ContractReader
+	// gets updated by the polling loop
+	chainConfigs map[cciptypes.ChainSelector]cciptypes.ChainConfig
+	// mapping between each node's peerID and the chains it supports. derived from chainConfigs
+	nodeSupportedChains map[libocrtypes.PeerID]mapset.Set[cciptypes.ChainSelector]
+	// set of chains that are known to CCIP, derived from chainConfigs
+	knownSourceChains mapset.Set[cciptypes.ChainSelector]
+	// map of chain to FChain value, derived from chainConfigs
+	fChain map[cciptypes.ChainSelector]int
+	lggr   logger.Logger
+	mutex  *sync.RWMutex
+
+	stopCh services.StopChan
+	services.StateMachine
+
+	// How frequent will the poller fetch the chain configs
+	pollingInterval time.Duration
+}
+
+func NewHomeChainConfigPoller(
+	homeChainReader types.ContractReader,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) *HomeChainConfigPoller {
+	return &HomeChainConfigPoller{
+		stopCh:              make(chan struct{}),
+		homeChainReader:     homeChainReader,
+		lggr:                lggr,
+		chainConfigs:        map[cciptypes.ChainSelector]cciptypes.ChainConfig{},
+		mutex:               &sync.RWMutex{},
+		nodeSupportedChains: map[libocrtypes.PeerID]mapset.Set[cciptypes.ChainSelector]{},
+		knownSourceChains:   mapset.NewSet[cciptypes.ChainSelector](),
+		fChain:              map[cciptypes.ChainSelector]int{},
+		pollingInterval:     pollingInterval,
+	}
+}
+
+func (r *HomeChainConfigPoller) Start(ctx context.Context) error {
+	err := r.fetchAndSetConfigs(ctx)
+	if err != nil {
+		r.lggr.Errorw("Initial fetch of on-chain configs failed", "err", err)
+		return fmt.Errorf("initial fetch of on-chain configs failed")
+	}
+	r.lggr.Infow("Start Polling ChainConfig")
+	return r.StartOnce(r.Name(), func() error {
+		go r.poll()
+		return nil
+	})
+}
+
+func (r *HomeChainConfigPoller) poll() {
+	ctx, cancel := r.stopCh.NewCtx()
+	defer cancel()
+	ticker := time.NewTicker(r.pollingInterval * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			err := r.fetchAndSetConfigs(ctx)
+			if err != nil {
```

Comment:
Style nit: saves one line

```suggestion
			if err := r.fetchAndSetConfigs(ctx); err != nil {
```